src/main/bugfree/Bugfree.php

```diff
@@ -45,7 +45,7 @@ public function __construct(Resolver $resolver, Config $config)
     public function parse($name, $source)
     {
         $result = new Result($name, $this->config);
-        $parser = new Parser(new Lexer());
+        $parser = (new ParserFactory())->create(ParserFactory::ONLY_PHP7, new Lexer());
```

Comment:
Possibly, but you may as well just fix your code so it can run on version 5 and 7. If you do that ONLY_PHP7 parses the code whether bugfree is running on php 5.5 or 7.0.

The only incompatibilities I've come across is a class that has the same name as a primitive and having a number that starts with a 0.

If you want to support both I don't mind changing it.